### PR TITLE
Support code based e-mail verification

### PR DESCRIPTION
**Pull request by alexaubry**
### Issues

With this PR, we add support for code-based e-mail registration in the modular authentication flow.

### Solutions

This is made possible by the changes in [sync engine](https://github.com/wireapp/wire-ios-sync-engine/pull/835). In the same spirit, we unite phone and e-mail verification event handlers and logic. 

### To-Do

- [x] Restore broken profile picture upload
- [x] Move alert strings to localized files

### CODE REVIEW

**Comment by vytis** [comment]
The diff would be much cleaner if you would merge develop into `feature/modular-authentication-flows` also.

File: Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme [hunk 1]
```diff
@@ -112,7 +112,7 @@
          </CommandLineArgument>
          <CommandLineArgument
             argument = "-ZMBackendEnvironmentType staging"
-            isEnabled = "NO">
+            isEnabled = "YES">
```
Comment (vytis) [inline Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme]: Probably needs to be reverted

File: Wire-iOS/Sources/UserInterface/Registration/Authentication/Coordinator/AuthenticationCoordinator.swift [hunk 2]
```diff
@@ -351,10 +412,10 @@ extension AuthenticationCoordinator {
         }
     }
 
-    func requestPhoneRegistration(with credentials: ZMPhoneCredentials, user: ZMIncompleteRegistrationUser) {
+    func requestPhoneRegistration(with credentials: ZMPhoneCredentials, user: UnregisteredUser) {
         presenter?.showLoadingView = true
-        transition(to: .validatePhoneIdentity(credentials: credentials, user: user))
-        unauthenticatedSession.verifyPhoneNumberForRegistration(credentials.phoneNumber!, verificationCode: credentials.phoneNumberVerificationCode!)
+//        transition(to: .validatePhoneIdentity(credentials: credentials, user: user))
+//        unauthenticatedSession.verifyPhoneNumberForRegistration(credentials.phoneNumber!, verificationCode: credentials.phoneNumberVerificationCode!)
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Registration/Authentication/Coordinator/AuthenticationCoordinator.swift]: Commented out code

File: Wire-iOS/Sources/UserInterface/Registration/Authentication/AuthenticationInterfaceBuilder.swift [hunk 3]
```diff
@@ -64,22 +63,30 @@ class AuthenticationInterfaceBuilder {
         case .noHistory(_, let type):
             return NoHistoryViewController(contextType: type)
 
-        case .verifyPhoneNumber(let phoneNumber, let user, _):
-            let verificationController = PhoneVerificationStepViewController(unregisteredUser: user)!
-            verificationController.phoneNumber = phoneNumber
-            return verificationController
+//        case .verifyPhoneNumber(let phoneNumber, _, _):
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Registration/Authentication/AuthenticationInterfaceBuilder.swift]: Commented out code

File: Wire-iOS/Sources/UserInterface/Registration/Authentication/Event Handlers/Registration/RegistrationFinalErrorHandler.swift [hunk 4]
```diff
@@ -29,8 +29,8 @@ class RegistrationFinalErrorHandler: AuthenticationEventHandler {
     func handleEvent(currentStep: AuthenticationFlowStep, context: NSError) -> [AuthenticationCoordinatorAction]? {
         let error = context
 
-        // Only handle final registration errors
-        guard case .finalizeRegistration = currentStep else {
+        // Only handle user cration errors
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Registration/Authentication/Event Handlers/Registration/RegistrationFinalErrorHandler.swift]: Typo - "cration"

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/SignIn/PhoneSignInViewController.m [hunk 5]
```diff
@@ -65,7 +65,7 @@ - (void)createPhoneNumberStepViewController
 {
     PhoneNumberStepViewController *phoneNumberStepViewController = [[PhoneNumberStepViewController alloc] init];
     phoneNumberStepViewController.view.translatesAutoresizingMaskIntoConstraints = NO;
-    phoneNumberStepViewController.delegate = self;
+    // phoneNumberStepViewController.delegate = self;
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/SignIn/PhoneSignInViewController.m]: Commented out code

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/TermsOfUseStepViewController.m [hunk 6]
```diff
@@ -103,7 +103,7 @@ - (void)createTermsOfUseText
 }
 
 - (void)onBackButtonPressed:(UIButton*)sender {
-    [[UnauthenticatedSession sharedSession] cancelWaitForEmailVerification];
+//    [[UnauthenticatedSession sharedSession] cancelWaitForEmailVerification];
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/TermsOfUseStepViewController.m]: Commented out code

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationTeam/Flow/TeamCreationFlowController.swift [hunk 7]
```diff
@@ -213,13 +213,13 @@ extension TeamCreationFlowController: VerifyEmailStepDescriptionDelegate {
 extension TeamCreationFlowController: SessionManagerCreatedSessionObserver {
     func sessionManagerCreated(userSession : ZMUserSession) {
         syncToken = ZMUserSession.addInitialSyncCompletionObserver(self, userSession: userSession)
-        let unauthenticatedSession = SessionManager.shared?.unauthenticatedSession
+        //let unauthenticatedSession = SessionManager.shared?.unauthenticatedSession
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationTeam/Flow/TeamCreationFlowController.swift]: Here as well

**Comment by alexaubry** [comment]
@vytis thanks for the feedback. I fixed the comments issues and the tests. I'll fix the login code related commented code in the next PR.

**Pull request merged**